### PR TITLE
Fix basemap attribution text disappearing

### LINKED ISSUE #3263: Basemap attribution text disappears on rotation and zoom for geopoint with maps and placement-map appearances
#### Software and hardware versions 
Collect v1.23.0

#### Problem description
#3255 added attribution text for basemap options from sources other than Mapbox and Google. These basemap options are displayed by the osmdroid mapping engine. Rotating the screen or zooming in to the current location makes these disappear.

#### Steps to reproduce the problem
1. Select a basemap source other than Mapbox or Google in General Settings > Maps > Basemap source.
1. Open a form such as All Widgets on the default server with geopoint questions with appearance maps and placement-map
1. Tap on Start GeoPoint on one of those questions
1. Notice that the attribution text is in the lower right corner
1. Rotate the screen
1. Notice that the attribution text is gone
1. Tap the back button
1. Tap on Start GeoPoint
1. Notice that the attribution text is in the lower right corner
1. Zoom to current position

#### Expected behavior
Attribution text should always be visible as it is with the geotrace and geoshape question types.

#### Other information 
This looks like a lifecycle management issue that may go deeper than just this.

### PR DESCRIPTION
**Pull request by nshestiuk**
Closes #3263

#### What has been done to verify that this works as intended?

I reproduced basemap attribution text disappearing on Android 7.1 (text disappears on rotation, zooming and also when the pointer is placed) and Android 9 (text disappears only when the pointer is placed) and confirmed that this PR fixes the issue on both.

**Android 9** (text disappears only when the pointer is placed)
with_issue
**Android 9** (fixed)
fixed

#### Why is this the best possible solution? Were any other approaches considered?

I've added missed code. That is obvious and simple solution.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

I do not see any risk of regression.

#### Do we need any specific form for testing your changes? If so, please attach one.
no
#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
no
#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/geo/OsmDroidMapFragment.java [hunk 1]
```diff
@@ -302,6 +302,7 @@
 
     @Override public void clearFeatures() {
         map.getOverlays().clear();
+        map.getOverlays().add(new AttributionOverlay(getContext()));
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/geo/OsmDroidMapFragment.java]: It looks like we have duplicated work we need to do to set up the view in both `clearFeatures` and `onCreateView` (this line and the line below for instance). I think it'd be good to extract this to a shared method so we're less likely to run into inconsistencies in the future. Does that make sense?

**Review by seadowg — CHANGES_REQUESTED** [review]

File: collect_app/src/main/java/org/odk/collect/android/geo/OsmDroidMapFragment.java (hunk 1 above)
Comment (nshestiuk) [reply collect_app/src/main/java/org/odk/collect/android/geo/OsmDroidMapFragment.java]: Done. 
If you have better idea for the method name, I whoud be happy to update PR

Comment (seadowg) [reply collect_app/src/main/java/org/odk/collect/android/geo/OsmDroidMapFragment.java]: 👍 Seems good for now!

**Comment by mmarciniak90** [comment]
**Tested with success**

Verified on Android: 4.2, 4.4, 5.1, 6.0, 7.0, 8.1, 9.0

Verified cases:
* OSM, USGS, Stamen, and CARTO maps
* zoom
* rotation
* GeoPoint widget
* GeoShape and GeoTrace - no regression noticed

@opendatakit-bot unlabel "needs testing"
@opendatakit-bot label "behavior verified"

**Review by seadowg — APPROVED** [review]

**Comment by lognaturel** [comment]
Nice to see you again, @nshestiuk, and thanks for taking this on!

**Pull request merged**